src/OpenSpaceToolkit/Core/Error/RuntimeError.cpp

```diff
@@ -11,23 +11,26 @@ namespace error
 
 RuntimeError::RuntimeError(const String& aMessage)
     : Exception(String::Empty()),
-      message_(aMessage)
+      message_(aMessage),
+      stackTrace_(boost::stacktrace::to_string(boost::stacktrace::stacktrace()))
 {
 }
 
-//                                 RuntimeError::RuntimeError                  (   const   String& aScope,
-//                                                                                 const   String& aMessage )
-//                                 :   Exception(aScope),
-//                                     message_(aMessage)
-// {
+String RuntimeError::getMessage() const
+{
+    return message_.data();
+}
 
-// }
+String RuntimeError::getStackTrace() const
+{
+    return stackTrace_.data();
+}
 
 RuntimeError::~RuntimeError() {}
 
 const char* RuntimeError::what() const noexcept
 {
-    return message_.data();
+    return (stackTrace_ + message_).data();
```

Comment:
Not sure this will work: `stackTrace_ + message_` is a local stack variable which you're returning a pointer to. That'll likely segfault.

Reply:
Solution is to store the `stackTrace_ + message_` as an instance variable, and return that instead.

Reply:
Hmmm... I see your point. It does pass without segfaulting in the tests, but easy enough to add an instance variable for safety.

Reply:
> It does pass without segfaulting in the tests

Could be because the memory hasn't been moved, since you're not doing anything in-between (there's no protection here, since you're using C pointers).

Reply:
updated